src/main/java/com/RecruitPlus/QuizPlatform/controller/QuestionController.java

```diff
@@ -48,4 +52,26 @@ public ResponseEntity<Object> deleteQuestion(@PathVariable String questionId){
         return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
 
     }
+    @GetMapping("/type/questions")
+    public List<Questions> getByType(@RequestParam String type)
+    {
+        List<Questions> ListOfQuestions=questionRepository.byType(type);
+        return ListOfQuestions;
+    }
+
+    @GetMapping("topics/questions")
+    public List<Questions> getByTopics(@RequestParam String[] topics)
+    {
+        List<Questions> QuestionsByCategory=questionRepository.byTopics(topics);
+        return QuestionsByCategory;
+    }
+
+    @GetMapping("/difficulty_level/questions")
+    public List<Questions> getByDifficultyLevel(@RequestParam String difficulty_level)
+    {
+        List<Questions> QuestionsByDiffLevel=questionRepository.byDiffLevel(difficulty_level);
+        return QuestionsByDiffLevel;
+    }
+
```

Comment:
These end points look fine. Can we try to see if we can implement this using one generic search endpoint?
Right now if i want to search both by type and topics, I'll have to do two api calls. Can we implement using one generic search endpoint?
An example end point might look like this.... /questions/v1/questions?topic=java&difficulty=easy

Reply:
yes we can do that...will work on that!